File: day3_mpi1/color/exercise/ex0a.c
#include <stdio.h>
#include <stdlib.h>
#include <mpi.h>

int main(int argc, char* argv[])
{

int nprocs, myid;

// Initialize MPI.
MPI_Init(&argc, &argv);
// Get size
MPI_Comm_size(MPI_COMM_WORLD, &nprocs);
// Get rank/id
MPI_Comm_rank(MPI_COMM_WORLD, &myid);

printf("Process %d has the color red.\n", myid);

printf("Process %d has the color green.\n", myid);

//Finalize
MPI_Finalize();
  return 0;

}

File: day3_mpi1/color/exercise/ex0a.f90
program pcolor 

implicit none

include "mpif.h"

integer :: nprocs, ierr, myid

! blank 1: Initialize MPI
call mpi_init(ierr)
! blank 2: Find the size and rank
call mpi_comm_size(mpi_comm_world, nprocs, ierr)
call mpi_comm_rank(mpi_comm_world, myid, ierr)

!print color
write(*,'("Process", I3, "has the color of red")') myid

write(*,'("Process", I3, "has the color of green")') myid

! blank 4: Finalize MPI
call mpi_finalize(ierr)

stop
end program pcolor

